docs/adr/ADR-002-tracking-module.md

```diff
@@ -0,0 +1,77 @@
+# ADR-002 Tracking Module
+
+## Status
+
+Already implemented.
+
+## Abstract
+
+We propose a new module called `x/tracking` that leverages our fork of [x/wasm](https://github.com/archway-network/archway-wasmd),
+which has contract gas logging capabilities.
+
+The module records contract gas usage on state, associating gas usage to the block and the tx in which the computational
+effort happened.
+
+## Context
+
+Archway requires an economically efficient way to track contract gas usage, in order to  create an environment in which
+contracts are rewarded fairly, deep inspection on what happens during `wasm` execution is required. Interactions between
+contracts need to be recorded and during each interaction we store the gas usage of a contract.
+
+Archway uses `gas` as a way to reward contracts. In fact `gas` is seen as contracts bringing volume into the network,
+and the network rewards.
```

Comment:
Archway uses `gas` as a way to reward contracts; In fact, it is the measure of contracts bringing volume to the network, and the protocol rewards accordingly.